packages/loader/container-loader/src/container.ts

```diff
@@ -1222,7 +1208,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.propagateConnectionState();
 
         // Internal context is fully loaded at this point
-        this.loaded = true;
+        this.lifecycleStateHandler.changeState("loaded");
```

Comment:
I'd suggest you to do the following:
1. rename lifecycleStateHandler to _lifecycleStateHandler
2. Expose lifecycleStateHandler getter & setter that will call methods on _lifecycleStateHandler
3. Move all the methods (like isLoaded(), isClosed() to Container object.

That way you can have same names with different meanings (as they are on different objects), i.e. Container.isLoaded() does not have to have same meaning as Container.lifecycleStateHandler === "loaded".
And the code becomes more streamlined / easier to read.